compiler/rustc_codegen_cranelift/src/base.rs

```diff
@@ -912,6 +912,16 @@ fn codegen_stmt<'tcx>(
             }
         }
         StatementKind::Coverage { .. } => fx.tcx.sess.fatal("-Zcoverage is unimplemented"),
+        StatementKind::CopyNonOverlapping(box rustc_middle::mir::CopyNonOverlapping {
+          src,
+          dst,
+          count,
+        }) => {
+            let dst = codegen_operand(fx, dst).load_scalar(fx);
+            let src = codegen_operand(fx, src).load_scalar(fx);
+            let count = codegen_operand(fx, count).load_scalar(fx);
+            fx.bcx.call_memcpy(fx.cx.module.target_config(), dst, src, count);
```

Comment:
Can you copy the implementation from https://github.com/rust-lang/rust/blob/5c688e52f19f8f0bb9e8c3a74119b96fdd1be07d/compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs#L524-L538

Reply:
I took a few liberties because I don't have direct access to the type, but it should be equivalent.